{{cookiecutter.project_name}}/.github/ISSUE_TEMPLATE/bug_report.yml

```diff
@@ -0,0 +1,71 @@
+name: Bug report
+description: Report something that is broken or incorrect
+labels: bug
+body:
+    - type: markdown
+      attributes:
+          value: |
+              **Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports)
+              detailing how to provide the necessary information for us to reproduce your bug. In brief:
+                * Please provide exact steps how to reproduce the bug in a clean Python environment.
+                * In case it's not clear what's causing this bug, please provide the data or the data generation procecure.
+                * Sometimes it is not possible to share the data but usually it is possible to replicate problems on publicly
+                  available datasets or to share a subset of your data.
+
+    - type: textarea
+      id: description
+      attributes:
+          label: Description of the bug
+          description: A clear and concise description of what the bug is.
+      validations:
+          required: true
+
+    - type: textarea
+      id: repex
+      attributes:
+          label: Minimal reproducible example
+          description: A code snipped that we can copy&paste to reproduce the bug without having any data
+          render: python
+          placeholder: |
+              import scanpy as sc
+              adata = sc.datasets.pbmc3k()
+
+              sc.do_something_that_doesnt_work(adata)
+
+    - type: textarea
+      id: stacktrace
+      attributes:
+          label: The error message produced by the code above
+          description: Please paste the entire error message
+          render: pytb
+          placeholder: |
+              ---------------------------------------------------------------------------
+              AssertionError                            Traceback (most recent call last)
+              <ipython-input-5-78750baef17a> in <module>
+              ----> 1 adata = sc.do_something_that_doesnt_work(adata)
+
+              /opt/conda/lib/python3.8/site-packages/scanpy/_preprocessing.py in do_something_that_doesnt_work(adata)
+                  438
+              --> 439     raise AssertionError("Demo function that fails")
+                  440
```

Comment:
I'm not sure this is a lot better than just having free-form markdown.

While many bug reports fit "ran this code, got this error" not all do. Especially difficult to solve ones. How does "incorrect thing works" or "this works, this works, this doesn't" fit into this template?

Reply:
All fields except for the first one are optional and you can write free-form markdown there.

Reply:
My preference here would be:

One required field with description of bug. This includes a template that looks like:

`````
## Description

## Example

```python
import {pkg}

# Minimal example to reproduce bug
````

```pytb
# Traceback from running code
```
`````

And a required field for version info.

To me, Text, Code block, Error block may be the most common bug report format, but I'm not sure it's over 50% prevalence. I think putting too much structure here makes it harder for people to report, since they'll try to fit their problem into structure.